### PR TITLE
feat: New Filter Conversations Menu - WPB-9440

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9440" title="WPB-9440" target="_blank">TaskWPB-9440</a>  [iOS] Create Context Menu for Filter Button
  </td></table>
  <br />
 

This PR introduces a new filter menu to the right navigation bar in YourViewController, along with a dynamic icon feature that changes the filter button icon when tapped. The changes enhance the user interface by providing a convenient way to filter conversations and visually indicate the state of the filter button.

## Changes

1. Filter Menu Addition:
- A new filter button is added to the right navigation bar, which displays a menu with multiple filtering options.
- The menu includes options for "All Conversations", "Favorites", "Groups", and "1:1 Conversations".
- Each menu option is represented by a UIAction with appropriate icons and titles.

2. Dynamic Icon for Filter Button:
- The filter button icon changes dynamically based on its selection state.
- When the filter button is tapped, the icon toggles between a filled and unfilled version to indicate its active state.

## Screenshot:

Screenshot 2024-05-27 at 09 29 50

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift [hunk 1]
```diff
@@ -112,20 +127,118 @@ extension ConversationListViewController {
...
+    private func getFilterImageName(for filter: FilterType, isSelected: Bool) -> String {
+        switch filter {
+        case .allConversations:
+            return isSelected ? FilterImageName.textBubbleFill.rawValue : FilterImageName.textBubble.rawValue
+        case .favorites:
+            return isSelected ? FilterImageName.starFill.rawValue : FilterImageName.star.rawValue
+        case .groups:
+            return isSelected ? FilterImageName.person3Fill.rawValue : FilterImageName.person3.rawValue
+        case .oneToOneConversations:
+            return isSelected ? FilterImageName.personFill.rawValue : FilterImageName.person.rawValue
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: Maybe one can return the right type and do the `.rawValue` once on the caller?

**Review by markusfassbender — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift [hunk 2]
```diff
@@ -112,20 +127,118 @@ extension ConversationListViewController {
...
+            self?.selectedFilter = filter
+            self?.setupRightNavigationBarButtons()
+        }
+
+        action.setValue(attributedTitle, forKey: "attributedTitle")
+
+        return action
+    }
+
+    private func getFilterImageName(for filter: FilterType, isSelected: Bool) -> String {
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: I'm wondering if we could extract this and maybe more of the new code from the `ConversationListViewController`.

**Review by caldrian — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: I was thinking the same thing. But for now decided to keep it here. In a follow up PR we can separate it into its own object.

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: Done! https://github.com/wireapp/wire-ios/commit/10529635d8ea8a54a6de63e16b863a87718d30b7

**Pull request merged**